docs/ref/tasks.txt

```diff
@@ -0,0 +1,303 @@
+    Some may use numbers, others UUIDs.
+    Task ids are always strings less than 64 characters.
+
+.. attribute:: TaskResult.status
+
+    The :class:`status <ResultStatus>` of the result.
+
+.. attribute:: TaskResult.enqueued_at
+
+    The time when the task was enqueued.
+
+    If :attr:`Task.enqueue_on_commit` was set, this is the time the transaction
+    completed.
+
+.. attribute:: TaskResult.started_at
+
+    The time when the task began execution.
+
+.. attribute:: TaskResult.finished_at
+
+    The time when the task finished execution, whether it failed or succeeded.
+
+.. attribute:: TaskResult.backend
+
+    The backend the result is from.
+
+.. attribute:: TaskResult.exception_class
+
+    The exception class raised when executing the task.
+
+    If the task has not finished, ``ValueError`` is raised. If the task finished
+    successfully, the exception class is ``None``.
+
+.. attribute:: TaskResult.traceback
+
+    The exception traceback from the raised exception when the task failed.
+
+    If the task has not finished, ``ValueError`` is raised. If the task finished
+    successfully, the traceback is ``None``.
+
+.. attribute:: TaskResult.return_value
+
+    The return value from the task function.
+
+    If the task has not finished yet, or failed, :exc:`ValueError` is raised.
+
+.. method:: TaskResult.refresh
+
+    Refresh the result's attributes from the queue store.
+
+.. method:: TaskResult.arefresh
+
+    The ``async`` variant of :meth:`TaskResult.refresh`.
+
+.. attribute:: TaskResult.is_finished
+
+    Whether the task has finished (successfully or not).
+
+Backends
+========
+
+Base backend
+------------
+
+.. module:: django.tasks.backends.base
+
+.. class:: BaseTaskBackend
+
+``BaseTaskBackend`` is the parent class for all task backends.
+
+.. method:: BaseTaskBackend.get_result(result_id)
+
+    Retrieve a result by its id. If the result does not exist,
+    :exc:`ResultDoesNotExist <django.tasks.exceptions.ResultDoesNotExist>`
+    is raised.
+
+    If the backend does not support ``get_result``, :exc:`NotImplementedError`
+    is raised.
+
+.. method:: BaseTaskBackend.aget_result(result_id)
+
+    The ``async`` variant of :meth:`BaseTaskBackend.get_result`.
+
+Features
+~~~~~~~~
+
+Some backends may not support all features Django provides. It's possible to
+identifying the supported functionality of a backend, and potentially
+changing behavior accordingly.
+
+.. attribute:: BaseTaskBackend.supports_defer
+
+    Whether the backend supports enqueueing tasks to be
+    executed after a specific time using the ``run_after`` attribute.
+
+.. attribute:: BaseTaskBackend.supports_async_task
+
+    Whether the backend supports enqueueing async functions (coroutines).
+
+.. attribute:: BaseTaskBackend.supports_get_result
+
+    Whether the backend supports retrieving task results from another thread
+    after they have been enqueued.
+
+Available backends
```

Comment:
Should each backend clarify what they support based on the list of supported task concepts (eg. `supports_defer: Yes|No`)